src/common/lib/util/defaults.ts

```diff
@@ -295,6 +295,17 @@ export function normaliseOptions(options: DeprecatedClientOptions): NormalisedCl
     }
   }
 
+  let internetAvailabilityParams = null;
+  if (options.internetAvailabilityUrl) {
+    let [uri, qs] = options.internetAvailabilityUrl.split('?');
+    const params = qs ? Utils.parseQueryString(qs) : {};
+    if (uri.indexOf('://') === -1) {
+      uri = (options.tls ? 'https://' : 'http://') + uri;
+    }
+    options.internetAvailabilityUrl = uri;
+    internetAvailabilityParams = params;
```

Comment:
Is it necessary to split out the query string and pass it around separately, only to join it again when making the request?